### PR TITLE
Client v2

### PR DESCRIPTION
**Pull request by cloderic**
- [x] Drop support for python v2
- [x] Remove six
- [x] Migrate to poetry
- [x] Get rid of deprecated functions/features
- [x] pandas v1
- [x] Rebase once #180 is merged
-----

Testable using the [release candidate](https://pypi.org/project/craft-ai/2.0.0rc5/) using `pip install craft-ai==2.0.0rc5`, `pip install craft-ai==2.0.0rc5[pandas]` (to get the pandas client)

### CODE REVIEW

**Comment by cloderic** [comment]
@yleprincecraft @sperretcraft @PaulineMCraft @juliencraftai please test and give feedback ! It's also the right time to make big changes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,26 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+  - `craft_ai.add_operations` becomes `craft_ai.add_agent_operations`,
+  - `craft_ai.add_operations_bulk` becomes `craft_ai.add_agents_operations_bulk`,
+  - `craft_ai.get_operations_list` becomes `craft_ai.get_agent_operations`,
+  - `craft_ai.get_generator_operations_list` becomes `craft_ai.get_generator_operations`,
+  - `craft_ai.get_state_history` becomes `craft_ai.get_agent_states`,
+  - `craft_ai.get_decision_tree` becomes `craft_ai.get_agent_decision_tree`,
+  - `craft_ai.get_decision_trees_bulk` becomes `craft_ai.get_agents_decision_trees_bulk`,
+  - `craft_ai.pandas.get_paths` becomes `craft_ai.collect_paths_from_tree`,
+  - `craft_ai.pandas.get_neighbours` becomes `craft_ai.compute_tree_decision_paths_neighbors`.
+
```
Comment (yleprincecraft) [inline CHANGELOG.md]: The previously called `_extract_tree` method should be Public

File: README.md [hunk 2]
```diff
@@ -1917,4 +1917,8 @@ addition_operations_bulk_payload = [
 
 client.add_operations_bulk(addition_operations_bulk_payload)
 ```
+<<<<<<< HEAD
```
Comment (yleprincecraft) [inline README.md]: Should be fixed :male_detective:

File: craft_ai/client.py [hunk 3]
```diff
@@ -0,0 +1,1107 @@
...
+    #########################
+    # Decision tree methods #
+    #########################
+
+    def _get_agent_decision_tree(
+        self, agent_id, timestamp, version=DEFAULT_DECISION_TREE_VERSION
+    ):
+        """Tool for the function get_agent_decision_tree.
+
+<<<<<<< HEAD
```
Comment (yleprincecraft) [inline craft_ai/client.py]: Should be fixed :female_detective:

File: craft_ai/pandas/client.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+                            for col in chunk.columns
+                            if is_valid_property_value(col, row[col])
+                        },
+                    }
+                    for _, row in chunk.iterrows()
+                ]
+                super(Client, self).add_agent_operations(agent_id, chunk_operations)
+
+            return {
+                "message": 'Successfully added %i operation(s) to the agent "%s/%s/%s" context.'
```
Comment (yleprincecraft) [inline craft_ai/pandas/client.py]: `%i` does not represent the real number of operations added :warning:

File: craft_ai/timezones.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    "AEST": "+10:00",
+    "AEDT": "+11:00",
+    "ACST": "+09:30",
+    "ACDT": "+10:30",
+    "AWST": "+08:00",
+}
+
+
+def is_timezone(value):
+    if isinstance(value, int) and value <= 840 and value >= -720:
```
Comment (yleprincecraft) [inline craft_ai/timezones.py]: Hard coded values `840` and `-720`

File: craft_ai/tree_utils.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    split = decision_path.split("-")
+    neighbours = []
+    for step in range(1, len(split) + 1):
+        for path in paths:
+            if _is_neighbour(path, "-".join(split[:step])):
+                neighbours.append(path)
+    return neighbours
+
+
+def _extract_tree(tree):
```
Comment (yleprincecraft) [inline craft_ai/tree_utils.py]: This method should be Public

File: CHANGELOG.md (hunk 1 above)
Comment (cloderic) [reply CHANGELOG.md]: done

File: README.md (hunk 2 above)
Comment (cloderic) [reply README.md]: This will be handled when we update the doc from the central repo

File: craft_ai/pandas/client.py (hunk 4 above)
Comment (cloderic) [reply craft_ai/pandas/client.py]: fixed !

File: craft_ai/tree_utils.py (hunk 6 above)
Comment (cloderic) [reply craft_ai/tree_utils.py]: fixed

File: craft_ai/timezones.py (hunk 5 above)
Comment (cloderic) [reply craft_ai/timezones.py]: I've added a comment for that

**Pull request merged**